Mathlib/LinearAlgebra/Matrix/ElementaryMatrix.lean

```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2023 Wen Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wen Yang
+-/
+import Mathlib.Data.Matrix.Basic
+
+/-!
+# Elementary Matrix
+-/
+
+namespace Matrix
+
+universe u v
+
+open Matrix
+
+variable {n : Type u} [DecidableEq n] {R : Type v}
+
+/-- A matrix with only one nonzero element.-/
+def Single [AddMonoid R] (i j : n) (c : R) : Matrix n n R := of fun (k1 : n) (k2 : n) =>
+  if k1 = i ∧ k2 = j then c
+  else 0
```

Comment:
This matrix already exists, it's [`Matrix.stdBasisMatrix`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Matrix/Basis.html#Matrix.stdBasisMatrix).